dialogue-core/src/main/java/com/palantir/dialogue/core/PinUntilErrorChannel.java

```diff
@@ -45,7 +45,9 @@
 import org.slf4j.LoggerFactory;
 
 /**
- * Routes all requests to one host until an error is received, then we'll move on to the next host.
+ * Routes all requests to one host until an error is received, then we'll move on to the next host. We explicitly
+ * do not consider limited responses as errors when determining whether to change hosts, trading off improved locality
+ * over overall throughput.
```

Comment:
We could expand on this to note that events which reduce the limit also result in changing the pinned channel.